src/components/controls/ColorMapControls.jsx

```diff
@@ -0,0 +1,53 @@
+import { Dropdown } from "react-bootstrap";
+
+import { useControlsContext } from "../../context/controls-context";
+
+export default function ColorMapControl(props) {
+  const {
+    state: { allColorMaps, colorMap },
+    dispatch,
+  } = useControlsContext();
+
+  return (
+    <div className="fullWidth">
+      <h4>Color Map</h4>
+      <Dropdown>
+        <Dropdown.Toggle variant="outline-primary" className="fullWidth">
+          <img
+            src={colorMap.src}
+            alt="Selected color map"
+            height="15"
+            width="65%"
+            className="mr-2"
+          />
+          {colorMap.name}
+        </Dropdown.Toggle>
+        <Dropdown.Menu>
+          {allColorMaps.map((color) => {
+            return (
+              <Dropdown.Item
+                key={color.name}
+                active={colorMap === color}
+                onClick={() =>
+                  dispatch({
+                    type: "CHANGE_COLOR_MAP",
+                    payload: color,
+                  })
+                }
```

Comment:
Don't be afraid to save stuff like this function into a variable. The variable will simplify the JSX portion of the component, and it won't create `n` copies of the function. Sometimes that matters when the function is sufficiently complicated